File: PomodoroLogic/PomodoroHelper.cs
using System;

namespace PomodoroLogic
{
    public class PomodoroHelper
    {
        public enum POMODORO_MODE
        {
            POMODORO,
            SHORT_BREAK,
            LONG_BREAK
        }

        public int getNewNumCompletedPoms(int prevNumCompletedPoms)
        {
            return prevNumCompletedPoms + 1;
        }

        public int getNewNumCompletedShortBreaks(int prevNumCompletedShortBreaks)
        {
            return prevNumCompletedShortBreaks + 1;
        }

        public int getNewNumCompletedLongBreaks(int prevNumCompletedLongBreaks)
        {
            return prevNumCompletedLongBreaks + 1;
        }

        public bool shouldStartLongBreak(int longBreakInterval, int numCompletedShortBreaks)
        {
            return numCompletedShortBreaks + 1 == longBreakInterval;
        }

        public POMODORO_MODE getCurrentPomodoroMode(
            int numCompletedPoms,
            int numCompletedShortBreaks,
            int numCompletedLongBreaks,
            int longBreakInterval)
        {
            if (numCompletedPoms == 0)
            {
                return POMODORO_MODE.POMODORO;
            }
            if (numCompletedPoms % longBreakInterval != 0)
            {
                if (numCompletedShortBreaks + numCompletedLongBreaks < numCompletedPoms)
                {
                    return POMODORO_MODE.SHORT_BREAK;
                }
                return POMODORO_MODE.POMODORO;
            } else if (numCompletedPoms / longBreakInterval == numCompletedLongBreaks)
            {
                return POMODORO_MODE.POMODORO;
            }
            return POMODORO_MODE.LONG_BREAK;
        }
    }
}

File: PomodoroApp/ClientApp/src/helpers/pomodoroHelper.js
export const POMODORO_MODES = Object.freeze({
  POMODORO: {
    type: Symbol('POMODORO'),
    time: {
      minutes: 25,
      seconds: 0,
    },
  },
  SHORT_BREAK: {
    type: Symbol('SHORT_BREAK'),
    time: {
      minutes: 5,
      seconds: 0,
    },
  },
  LONG_BREAK: {
    type: Symbol('LONG_BREAK'),
    time: {
      minutes: 15,
      seconds: 0,
    },
  },
});

export const getCurrentPomodoroMode = (
  numCompletedPoms,
  numCompletedShortBreaks,
  numCompletedLongBreaks,
  longBreakInterval,
) => {
  if (numCompletedPoms === 0) {
    return POMODORO_MODES.POMODORO.type;
  }
  if (numCompletedPoms % longBreakInterval !== 0) {
    if (numCompletedShortBreaks + numCompletedLongBreaks < numCompletedPoms) {
      return POMODORO_MODES.SHORT_BREAK.type;
    }
    return POMODORO_MODES.POMODORO.type;
  } else if (numCompletedPoms / longBreakInterval === numCompletedLongBreaks) {
    return POMODORO_MODES.POMODORO.type;
  }
  return POMODORO_MODES.LONG_BREAK.type;
};

export const getNextPomodoroMode = (
  currentMode,
  numCompletedPoms,
  numCompletedShortBreaks,
  longBreakInterval,
) => {
  switch (currentMode) {
    case POMODORO_MODES.POMODORO.type:
      if ((numCompletedPoms + 1) % longBreakInterval !== 0) {
        if (numCompletedShortBreaks < (numCompletedPoms + 1)) {
          return POMODORO_MODES.SHORT_BREAK.type;
        }
        return POMODORO_MODES.POMODORO.type;
      }
      return POMODORO_MODES.LONG_BREAK.type;
    case POMODORO_MODES.SHORT_BREAK.type:
    case POMODORO_MODES.LONG_BREAK.type:
    default:
      return POMODORO_MODES.POMODORO.type;
  }
};
